services/submission/src/main/java/app/coronawarn/server/services/submission/validation/ValidSubmissionPayload.java

```diff
@@ -190,21 +151,37 @@ private boolean checkKeysCumulateEqualOrLessThanMaxRollingPeriodPerDay(List<Temp
 
     private boolean keysHaveFlexibleRollingPeriod(List<TemporaryExposureKey> exposureKeys) {
       return exposureKeys.stream()
-          .anyMatch(temporaryExposureKey ->
-              temporaryExposureKey.getRollingPeriod() < submissionServiceConfig.getMaxRollingPeriod());
+          .anyMatch(temporaryExposureKey -> temporaryExposureKey.getRollingPeriod() < maxRollingPeriod);
     }
 
     private boolean checkStartIntervalNumberIsAtMidNight(List<TemporaryExposureKey> exposureKeys,
         ConstraintValidatorContext validatorContext) {
       boolean isNotMidNight00Utc = exposureKeys.stream()
-          .anyMatch(exposureKey ->
-              exposureKey.getRollingStartIntervalNumber() % submissionServiceConfig.getMaxRollingPeriod() > 0);
+          .anyMatch(exposureKey -> exposureKey.getRollingStartIntervalNumber() % maxRollingPeriod > 0);
 
       if (isNotMidNight00Utc) {
         addViolation(validatorContext, "Start Interval Number must be at midnight ( 00:00 UTC )");
         return false;
       }
+
       return true;
     }
+
+    private boolean checkVisitedCountriesAreValid(SubmissionPayload submissionPayload,
+        ConstraintValidatorContext validatorContext) {
+      List<String> supportedCountriesList = List.of(supportedCountries);
+      AtomicBoolean validCountries = new AtomicBoolean(true);
+
+      submissionPayload.getVisitedCountriesList().forEach(country -> {
+        if (!supportedCountriesList.contains(country)) {
```

Comment:
Since we are comparing strings, we should be careful. I'm wondering if the scenarios below could happen:
 - Payload might contain lower/upper case -> maybe do ignore case checks
 - Payload might contain spaces in between comma separated countries ? -> maybe trim the input before comparison

Also, in case a payload contains a lot of valid countries, but one unsupported country ('DE,FR, AA' ), what should be our strategy? Do we reject the payload or maybe make it more resilient and accept only the valid country codes? Could be the case that one country was dropped from the supported list, but that doesn't mean that we should ignore the fact that the user travelled to the other supported countries.